src/js/angular/similarity/controllers/create-index.controller.js

```diff
@@ -12,741 +24,679 @@ CreateSimilarityIdxCtrl.$inject = ['$scope', 'toastr', '$uibModal', '$timeout',
-        // if we execute a CONSTRUCT and then a SELECT. This makes sure it's always table.
-        $scope.currentQuery.outputType = 'table';
-        $scope.runQuery();
+    /**
+     * Validates all similarity index data, and if everything is ok calls backend server to create an index.
+     */
+    $scope.createSimilarityIndex = () => {
+        $scope.saveOrUpdateExecuted = true;
+        updateQueryFromEditor($scope.similarityIndexInfo)
+            .then(validateSimilarityIndex)
+            .then(validateSimilarityIndexNameExistence)
+            .then((similarityIndexInfo) => createIndex(similarityIndexInfo.getSimilarityIndex()))
+            .catch((error) => {
+                if (!(error instanceof SimilarityIndexError)) {
+                    toastr.error(getError(error), $translate.instant('similarity.could.not.get.indexes.error'));
+                }
+            });
     };
 
-    $scope.resetCurrentTabConfig = () => {
-        $scope.currentTabConfig = {
-            pageSize: 100, // page limit 100 as this is only used for preview
-            page: 1,
-            allResultsCount: 0,
-            resultsCount: 0
-        };
-    };
+    /**
+     * Opens a modal dialog and displays the sparql query of currently selected index type query.
+     */
+    $scope.viewQuery = () => {
+        updateQueryFromEditor($scope.similarityIndexInfo)
+            .then(validateSimilarityIndex)
+            .then(fetchSparqlQuery)
+            .then(showSparqlQuery)
+            .catch((error) => toastr.error(getError(error)));
+    };
+
+    /**
+     * Executes the query.
+     */
+    $scope.preview = () => {
+        updateQueryFromEditor($scope.similarityIndexInfo)
+            .then(validateQueryType)
+            .then(validateQuery);
+
+        getOntotextYasgui().query();
+        // TODO render can be through method instead change all configuration
+        $scope.yasguiConfig = {...$scope.yasguiConfig, render: RenderingMode.YASR};
+    }
 
-    // query tab operations
-    $scope.saveTab = (id) => {
-        const idx = findTabIndexByID(id);
-        // Tab was deleted, don't try to save it's state
-        if (idx === undefined) {
-            return {};
-        }
-        const tab = $scope.tabsData[idx];
-        //tab.query = window.editor.getValue();
-        $scope.saveQueryToLocal(tab);
-        return tab;
+    $scope.saveSearchQuery = function () {
+        updateQueryFromEditor($scope.similarityIndexInfo)
+            .then(validateSimilarityIndexName)
+            .then((similarityIndexInfo) => {
+                const isSearchQuery = similarityIndexInfo.isSearchTypeSelected();
+                let data = {
+                    name: similarityIndexInfo.getSimilarityIndex().name,
+                    changedQuery: similarityIndexInfo.getQuery(),
+                    isSearchQuery
+                };
+                return SimilarityRestService.saveSearchQuery(JSON.stringify(data)).then(() => isSearchQuery)
```

Comment:
This looks strange. Can you try flattening this?

Reply:
Done